src/script/lua_api/l_item.cpp

```diff
@@ -211,8 +225,6 @@ int LuaItemStack::l_to_table(lua_State *L)
 		lua_setfield(L, -2, "count");
 		lua_pushinteger(L, item.wear);
 		lua_setfield(L, -2, "wear");
-		lua_pushlstring(L, item.metadata.c_str(), item.metadata.size());
-		lua_setfield(L, -2, "metadata");
```

Comment:
Do you know any mod that uses metadata inside itemstack:to_table()? If any, then they probably don't access this field but only pass it back to the itemstack constructor, which handles it.